cylc/flow/etc/job.sh

```diff
@@ -29,9 +29,21 @@ cylc__job__main() {
     if "${CYLC_DEBUG:-false}"; then
         if [[ -n "${BASH:-}" ]]; then
             PS4='+[\D{%Y%m%dT%H%M%S%z}]\u@\h '
-            exec 19>>"${CYLC_SUITE_RUN_DIR}/log/job/${CYLC_TASK_JOB}/job.xtrace"
-            export BASH_XTRACEFD=19
-            >&2 echo "Sending DEBUG MODE xtrace to job.xtrace"
+            X_Y=${BASH_VERSION:0:3} # e.g. bash 4.2
+            if (( ${X_Y:0:1} < 4 )); then
+                # bash-3 (old!)
+                >&2 echo "WARNING: can't send DEBUG MODE xtrace output to job.xtrace in bash-$X_Y"
+            else
+                >&2 echo "Sending DEBUG MODE xtrace output to job.xtrace"
+                if [[ "$X_Y" == "4.0" ]]; then
```

Comment:
Although I've found various comments online implying this works at 4.0, the bash changelog implies this was introduced in 4.1: https://github.com/bminor/bash/blob/6c6454cb18d7cd30b3b26d5ba6479431e599f3ed/CHANGES#L3396

Reply:
Another reference which says this came in at 4.1: https://wiki.bash-hackers.org/scripting/bashchanges#builtin_variables